Sources/FoundationEssentials/FileManager/FileManager+SymbolicLinks.swift

```diff
@@ -40,14 +42,35 @@ extension _FileManagerImpl {
         guard !destPath.isEmpty else {
             throw CocoaError.errorWithFilePath(.fileNoSuchFile, destURL)
         }
-        
-        try createSymbolicLink(atPath: path, withDestinationPath: destPath)
+
+#if os(Windows)
+        let bIsDirectory = try path.withNTPathRepresentation {
+            var faAttributes: WIN32_FILE_ATTRIBUTE_DATA = .init()
+            guard GetFileAttributesExW($0, GetFileExInfoStandard, &faAttributes) else {
+                throw CocoaError.errorWithFilePath(path, win32: GetLastError(), reading: true)
+            }
+            return faAttributes.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY == FILE_ATTRIBUTE_DIRECTORY
+        }
+#else
+        let bIsDirectory = false
+#endif
+        try createSymbolicLink(atPath: path, withDestinationPath: destPath, isDirectory: bIsDirectory)
     }
     
     func createSymbolicLink(
         atPath path: String,
-        withDestinationPath destPath: String
+        withDestinationPath destPath: String,
+        isDirectory: Bool
```

Comment:
This function gets called from the URL function above, the API in `SwiftFileManager.swift`, and the API in `FileManager+Bridge.swift`. Instead of duplicating the `isDirectory` check to each of those places, could we just calculate it in this function before we call `CreateSymbolicLinkW`?

Reply:
Yeah, I think that we should be able to sink it here.